Packs/HarfangLabEDR/Integrations/Hurukai/Hurukai.py

```diff
@@ -0,0 +1,2284 @@
+
+    readable_output = tableToMarkdown(f'Endpoint information for Hostname : {hostname}', data['results'])
+
+    outputs = {
+        'Harfanglab.Agent(val.agentid == obj.agentid)': data['results']
+    }
+
+    return_outputs(
+        readable_output,
+        outputs,
+        data
+    )
+    return data
+
+
+def job_create(client, args, parameters=None, can_use_previous_job=True):
+    action = args.get('action', None)
+    agent_id = args.get('agent_id', None)
+
+    if action is None or agent_id is None:
+        return False, None
+
+    if can_use_previous_job:
+        previous_job_id = find_previous_job(client, action, agent_id)
+        if previous_job_id is not None:
+            return True, previous_job_id
+
+    data = client.job_create(agent_id, action, parameters)
+    time.sleep(2)
+
+    job_id = data[0]['id']
+    return True, job_id
+
+
+"""
+    Returns a job status (context dict)
+"""
+
+
+def get_job_status(client, job_id):
+
+    info = client.job_info(job_id)
+
+    status = "running"
+
+    if info['instance'] == info['done']:
+        status = "finished"
+    elif info['error'] > 0:
+        status = "error"
+    elif info['canceled'] > 0:
+        status = "canceled"
+    elif info['waiting'] > 0:
+        status = "waiting"
+    elif info['running'] > 0:
+        status = "running"
+    elif info['injecting'] > 0:
+        status = "injecting"
+
+    # Creation time formating
+    time_info = info['creationtime'].split('.')
+    time_info = time_info[0].replace('T', ' ').replace('Z', ' ')
+
+    context = {
+        'ID': job_id,
+        'Status': status,
+        'Creation date': time_info
+    }
+    return context
+
+
+def job_info(client, args):
+    # ret vals : starting, running, finished
+    job_ids = argToList(str(args.get('ids', None)))
+
+    context = []
+    for job_id in job_ids:
+        context.append(get_job_status(client, job_id))
+
+    ec = {
+        'Harfanglab.Job.Info(val.ID && val.ID == obj.ID)': context,
+    }
+    readable_output = tableToMarkdown('Jobs Info', context, headers=['ID', 'Status', 'Creation date'])
+
+    entry = {
+        'Type': entryTypes['note'],
+        'Contents': context,
+        'ContentsFormat': formats['json'],
+        'ReadableContentsFormat': formats['markdown'],
+        'HumanReadable': readable_output,
+        'EntryContext': ec
+    }
+
+    demisto.results(entry)
+    return context
+
+
+def find_previous_job(client, action, agent_id):
+    starttime = (datetime.now(timezone.utc) - timedelta(minutes=5)).strftime('%Y-%m-%d %H:%M')
+    args = {
+        'agent_id': agent_id,
+        'action': action,
+        'state': 2,
+        'ordering': '-starttime',
+        'starttime__gte': starttime
+    }
+    data = client.jobinstance_list(args)
+    job_id = None
+    if data['count'] > 0:
+        job_id = data['results'][0]['job_id']
+
+    return job_id
+
+
+def common_result():
+    # temporary, data need to reach ES
+    time.sleep(10)
```

Comment:
Please remove this. What is it used for?

Reply:
Rejected. We may have a latency between the job status modification time and the availability of job results.

Reply:
`time.sleep()` statements are not allowed in our code, so we will have to find a different solution.
If it can be done inside the playbook- it will be better. 
Are all these calls necessary, or that some of them can be removed?

Reply:
I have to modify 20 playbooks then. :-( 
No exception to this rule ?

Reply:
Are those playbooks using pooling? If so, there is no need to use `sleep` as it does the same thing.
You can try to remove it and see if anything changes.

Why will you update the generic polling in the next version of the connector? It can also solve it.

Reply:
Hi @Pierre-HarfangLab, have you tried the solution I suggested?

Reply:
Hi Adi,
I do not understand why generic polling and sleep do the same thing. In my understanding:
- generic polling is looping until the filter of the polling command output returns an empty string
- sleep is just a way to wait for an absolute amount of time. 
In our case, we need to wait x seconds after the generic polling task finishes. 

Using the XSOAR sleep task seems to be the proper way to do it. 
Do you confirm ?

Thanks.

Reply:
The [scheduled commands](https://xsoar.pan.dev/docs/integrations/scheduled-commands) can solve it, but right now it will be fine and we will merge it this way.